osu.Game/Online/API/Requests/GetUserMostPlayedBeatmapsRequest.cs

```diff
@@ -0,0 +1,66 @@
+﻿// Copyright (c) 2007-2017 ppy Pty Ltd <contact@ppy.sh>.
+// Licensed under the MIT Licence - https://raw.githubusercontent.com/ppy/osu/master/LICENCE
+
+using System;
+using System.Linq;
+using Newtonsoft.Json;
+using osu.Game.Beatmaps;
+using osu.Game.Rulesets;
+
+namespace osu.Game.Online.API.Requests
+{
+    public class GetUserMostPlayedBeatmapsRequest : GetUserBeatmapsRequest<UserMostPlayedBeatmapsResponse>
+    {
+        public GetUserMostPlayedBeatmapsRequest(long userID, BeatmapSetType type, int offset = 0)
+            : base(userID, type, offset)
+        {
+            if (type != BeatmapSetType.MostPlayed)
+                throw new ArgumentException("Please use " + nameof(GetUserBeatmapsRequest) + " instead");
+        }
+    }
+
+    public class UserMostPlayedBeatmapsResponse

```

Comment:
rather than doing this, i'm pretty sure you want to derive from the existing `BeatmapInfo` class.

Reply:
why? none of the properties match up. `BeatmapResponse` could be a BeatmapInfo

Reply:
BeatmapResponse should also be a derived BeatmapInfo, yes.

Reply:
why should is derive from BeatmapInfo?

Reply:
okay so you're clearly not understanding what is going on here.

you should not be deriving from `GetUserBeatmapsRequest` because the response is completely different. it should be its own stand-alone request. 

as a result you don't need to pollute the enum type of the other request, nor have weird exceptions thrown if you attempt to use the wrong request.